### ISSUE TITLE
Misuse of Parent Classes and Unnecessary in Code for Encoding? Obfuscated Function Call

### ISSUE THREAD

**Issue by GangGreenTemperTatum** [leondz/garak#134]
Code recommendation 🙂 from attempting to understand what prompts are actually being sent to the API

> ❗ Lines indented are to provide context on reasoning

**Problem:**

- The `InjectQP` class has a "parent class" of `TextProbe`. 
> In Python, a parent class is a way to basically define some basic functionality, and then the children can replace or add to the functionality from the parent. That's what the (TextProbe) means in this context.
- Looking at the `TextProbe` [definition](https://github.com/leondz/garak/blob/main/garak/probes/base.py#L73), it calls a "**super**" function. 
> Super is how you refer to the parent implementation. You can also define what the parent does. So `InjectQP` does **NOT** define a prompt function, which means it just falls back to the parent, `TextProbe`. And `TextProbe` has its own parent, `Probe`. `Probe` has all of the implementation that actually does work as defined [here](https://github.com/leondz/garak/blob/main/garak/probes/base.py#L43)
- So what's `InjectQP` value contributing here? `InjectQP` has an `init` function. The `init` function sets **WHICH** settings get used when you go up the parent chain.
- For example, in the top most parent, `Probe`, you can see that the probe function operates on a variable called `self.prompts` [here](https://github.com/leondz/garak/blob/main/garak/probes/base.py#L48)
- If you look anywhere else in that file, nothing sets the value of `self.prompts`, which comes from [here](https://github.com/leondz/garak/blob/a437307913ef08f39e1911cb2d631da59f472dea/garak/probes/encoding.py#L309)
- **TL;DR:** The contents of self prompts is set in the init function of `InjectQP`.

**Recommendation:**

- `TextPrompt` (`InjectQP` & `Probe`) has two functions that both call parent and nothing else. 
- Why not have `InjectQP` just have Prompt as it's parent instead of there being two steps?

**Additional References:**

- [Stop writing classes - YouTube](https://youtu.be/o9pEzgHorH0)

@leondz if you want to look into this further, happy to AD HOC via Slack etc 👍🏼

**Comment by leondz** [comment]
> I actually piped the reports with gron and jq which makes analysis much easier. Maybe another suggestion for the report output?

Would love to hear more. Could you open another issue, under e.g. "improve reporting format" or "report JSON wishes" or something like that, and say what would make life easier in there?

**Comment by leondz** [comment]
PR sounds fine, sure!

**Comment by leondz** [comment]
TextProbe factored out in 41a55a9c42836b2c76d3a394807b98924d65f077

**Comment by leondz** [comment]
spurious probe constructors pruned in 4200e44d3d1a49245086117784390893663f27eb

I think this wraps up the refactoring for now - if anything was missed, please just reopen & comment :)
